### DISCUSSION TITLE
docs/k3s-installation/

### DISCUSSION THREAD

**Discussion by giscus** [ricsanfre/pi-cluster#142] [Announcements]
# docs/k3s-installation/

How to install K3s, a lightweight kubernetes distribution, in our Pi Kuberentes cluster. Single master node and high availability deployment can be used.

https://picluster.ricsanfre.com/docs/k3s-installation/

**Comment by zlinuxboy** [comment]
well, not familiar with this comment system.
It's a wonderful blog about k3s on pi. I am on the road to deploy k3s on odroid xu4 cluster(similar to raspberry pi). according to your post.
1. create /etc/rancher/k3s/config.yaml
2. curl -sfL https://get.k3s.io | sh -s - server --cluster-init
but after install k3s, the `/etc/rancher/k3s/` only contain k3s.yaml, the config.yaml has gone. so my question is does it apply to the k3s installation? what happen to it?

**Reply by ricsanfre** [reply]
Hi @zlinuxboy,

I am not familiar with odroid xu4 SBCs, but i suppose that, as with the raspberry pis, some kind of kernel tuning (enabling cgroups) is required to make kubernetes work. Are you using Ubuntu 22.04? Is k3s.server generating any logs?

Regards
Ricardo

**Comment by neu7ron2** [comment]
Hi Ricardo
As Kubernetes is changing so frequently there are some updates to this page: 
The recommended way to taint the master node is with: node-role.kubernetes.io/control-plane:NoSchedule
(node-role.kubernetes.io/master=true:NoSchedule is deprecated)
Also you don't need to set the value to '=true'  for this control-pain taint as recommended in the documentation https://kubernetes.io/docs/reference/labels-annotations-taints/ 
In fact the '=true' value can cause some issues with 3rd party apps not supplying the value when adding tolerations)

Regards
Paul

**Reply by ricsanfre** [reply]
Thanks Paul

Updated Ansible's playbooks and this documentation page to use the recommended way to taint the master node.

 Also fluent-bit helm deployment and corresponding documentation page (https://picluster.ricsanfre.com/docs/logging-forwarder-aggregator/), have been updated, so new taint can be used when specifying Pod's tolerations.

Regards
Ricardo

**Comment by benbourner** [comment]
The install k3s command for the server is not consistent with the documentation. one says --disable 'local-storage', the other --disable 'local-path'

**Reply by ricsanfre** [reply]
Thanks @benbourner.

Inconsistency fixed.

**Comment by Ludwig1770** [comment]
I'm a bit puzzled on the "Master node installation" section of "Single-server Setup with an Embedded DB".  The instructions state:
  curl -sfL https://get.k3s.io | K3S_TOKEN=<server_token> ...
but looking at the k3s documentation, the server token is an artefact of the installation of the k3s server (in /var/lib/rancher/k3s/server/node-token) - is this not a chicken/egg situation?

**Reply by ricsanfre** [reply]
Hi @Ludwig1770, it is not a chicken/egg situation. If k3s token is not provided to installation script through K3S_TOKEN environment, installation process will generate one for you and store a it at `/var/lib/rancher/k3s/server/node-token`

In my automated deployment I generate all secrets needed by the deployment in advance. If you check my automated installation code using ansible, I set all K3s installation parameters using a configuration installed in every node (see k3s doc: https://docs.k3s.io/installation/configuration#configuration-file), instead of command line arguments and environment variables. You can check ansible playbook here: https://github.com/ricsanfre/pi-cluster/blob/master/ansible/k3s_install.yml

**Reply by Ludwig1770** [reply]
Thank you Sir
